src/alire/alire-builds.adb

```diff
@@ -0,0 +1,135 @@
+with Alire.Properties.Actions.Executor;
+with Alire.Utils.Tools;
+
+package body Alire.Builds is
+
+   use Directories.Operators; -- "/"
+
+   ----------------------------
+   -- Sandboxed_Dependencies --
+   ----------------------------
+
+   function Sandboxed_Dependencies return Boolean
+   is (not Config.DB.Get (Config.Keys.Dependencies_Shared,
+                          Config.Defaults.Dependencies_Shared));
+
+   -------------------
+   -- To_Msys2_Path --
+   -------------------
+   --  Convert C:\blah\blah into /c/blah/blah. This is needed because otherwise
+   --  rsync confuses drive letters with remote hostnames. This might be useful
+   --  in our troubles with tar?
+   function To_Msys2_Path (Path : Absolute_Path) return String
+   is
+   begin
+      if not Platforms.Current.On_Windows then
+         return Path;
+      end if;
+
+      declare
+         use AAA.Strings;
+         New_Path : String (1 .. Path'Length) := Path;
+      begin
+         --  Replace ':' with '/'
+         New_Path (2) := '/';
+
+         --  Replace '\' with '/'
+         return "/" & Replace (New_Path, "\", "/");
+      end;
+   end To_Msys2_Path;
+
+   ----------
+   -- Sync --
+   ----------
+
+   procedure Sync (Release   : Releases.Release;
+                   Was_There : out Boolean)
+   is
+      Src       : constant Absolute_Path := Paths.Vault.Path
+                                            / Release.Deployment_Folder;
+      Dst       : constant Absolute_Path := Builds.Path (Release);
+      Completed : Directories.Completion := Directories.New_Completion (Dst);
+
+      use AAA.Strings;
+   begin
+      Was_There := False;
+
+      if Completed.Is_Complete then
+         Trace.Detail ("Skipping build syncing to existing " & Dst);
+         Was_There := True;
+         return;
+      end if;
+
+      Directories.Create_Tree (Dst);
+      --  Ensure the sync destination dir exists
+
+      Utils.Tools.Check_Tool (Utils.Tools.Rsync);
+
+      declare
+         Busy : constant Simple_Logging.Ongoing :=
+                  Simple_Logging.Activity
+                    ("Syncing " & Release.Milestone.TTY_Image)
+                    with Unreferenced;
+      begin
+         OS_Lib.Subprocess.Checked_Spawn
+           (Command   => "rsync",
+            Arguments =>
+              To_Vector ("--del")
+            & "-aChH"
+            --  Archive, no CVS folders, keep hard links, human units
+            & (if Log_Level > Detail then To_Vector ("-P") else Empty_Vector)
+            & (if Log_Level < Info   then To_Vector ("-q") else Empty_Vector)
+            --  Trailing '/' to access contents directly in the following paths
+            & To_Msys2_Path (Src / "")
+            & To_Msys2_Path (Dst / "")
+           );
+         --  TODO: this may take some time, and rsync doesn't have a way
+         --  to show oneliner progress, so at some point we may want to use
+         --  something like GNAT.Expect for our spawns, and spin on newlines
```

Comment:
either spawn or [GNATCOLL.OS.Process](https://github.com/AdaCore/gnatcoll-core/blob/master/src/os/gnatcoll-os-process.ads) - spawn is higher level and suited for integration in a main loop. GNATCOLL.OS.Process is lower level and leaves the user more things to do, but is an easier in-place replacement for GNAT.Expect. Both are superior to GNAT.Expect since they allow separating stdout and stderr, and allow controlling the environment of the spawned process without affecting the parent's environment.